src/cow-react/modules/limitOrders/containers/OrdersWidget/index.tsx

```diff
@@ -21,6 +22,7 @@ export function OrdersWidget() {
   const history = useHistory()
   const ordersList = useLimitOrdersList()
   const { chainId, account } = useWeb3React()
+  const cancelOrder = useCancelOrder()
 
   const [cancelModalProps, setCancelModalProps] = useState<CancellationModalProps | null>(null)
```

Comment:
probably we will want to move this state to global state? or this is sth we only will do from the orders widget?

Reply:
This is more of a question for @shoom3301 and @nenadV91.
Another point is that the way it's implemented for limit orders currently, it doesn't handle the error or the signing case.
Should it?

Reply:
I would like to have generic solutions as simple as possible.
I mean, something like order cancelation should provide as simple as possible API with a minimum of integration points. Like:

```
const cancelOrder = useCanelOrder()

...

<button onClick={() => cancelOrder(orderId)}>
```

And that hook should render modal, initiate signing of typed data and so on.

Reply:
What about something like this?

```
// This should:
// 1. export cancelOrder
// 2. take in as parameter only the order id (maybe at the hook level?)
// 3. identify what type of cancellation is possible based on order & wallet type
//   3.1 expose as part of the hook if the order can be cancelled
// 4. render cancellation modal on callback trigger
// 5. handle signing
// 6. handle errors
// 7. update state

export type UseCancelOrderProps = {
  orderId: string
}

export type UseCancelOrderReturn = {
  isCancellable: boolean
  cancelOrder: () => void
}

export function useCancelOrder(props: UseCancelOrderProps): UseCancelOrderReturn {}
```

Reply:
Looks great!